Userland/Utilities/gunzip.cpp

```diff
@@ -41,10 +41,13 @@ ErrorOr<int> serenity_main(Main::Arguments args)
         keep_input_files = true;
 
     for (auto filename : filenames) {
-        if (!filename.ends_with(".gz"))
-            filename = String::formatted("{}.gz", filename);
 
-        const auto input_filename = filename;
+        const auto input_filename = [filename]() -> String {
+            if (!filename.ends_with(".gz"))
+                return String::formatted("{}.gz", filename);
+            return filename;
+        }();
```

Comment:
What's the point of the immediately called lambda here? wouldn't this just work:
```suggestion
        String input_filename;
        if (filename.ends_with(".gz"))
            input_filename = filename;
        else
            input_filename = String::formatted("{}.gz", filename);
```

Reply:
It works, but the lambda improve const-correctness without any computational overhead.

Reply:
> without any computational overhead
Is that based on data, or a guess? But putting that aside, having a const variable for a string (which are immutable anyways) does not seem to outweigh the cost of making the code more complicated like that.

Reply:
I obviously don't have any data for this particular case.
I can't retrieve the source, but I'm quite sure I got the idea from a Jason Turner's talk at CppCon.
Immutability of String doesn't prevent an assignment, whereas the const qualifier does.
Anyway, if you find it much more complicated, I will change it.

Reply:
I guess it's fine, the majority of the serenity code base does not use const variables for rvalues, usually just for references and pointers, which is why I don't see the point of overcomplicating the code just to get that const in there, but I'll take your word for it that it doesn't have any adverse effects :shrug: